test/built-ins/AsyncGeneratorPrototype/next/request-queue-await-order.js

```diff
@@ -0,0 +1,59 @@
+// Copyright (C) 2018 Valerie Young. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-asyncgenerator-prototype-next
+description: next() requests from iterator processed in order, await
+info: >
+  AsyncGenerator.prototype.next ( value )
+  1. Let generator be the this value.
+  2. Let completion be NormalCompletion(value).
+  3. Return ! AsyncGeneratorEnqueue(generator, completion).
+
+  AsyncGeneratorEnqueue ( generator, completion )
+  ...
+  4. Let queue be generator.[[AsyncGeneratorQueue]].
+  5. Let request be AsyncGeneratorRequest{[[Completion]]: completion,
+     [[Capability]]: promiseCapability}.
+  6. Append request to the end of queue.
+  ...
+
+  AsyncGeneratorResolve ( generator, value, done )
+  ...
+  2. Let queue be generator.[[AsyncGeneratorQueue]].
+  3. Assert: queue is not an empty List.
+  4. Remove the first element from queue and let next be the value of that element.
+  ...
+
+flags: [async]
+features: [async-iteration]
+---*/
+
+var yieldorder = 0;
+
+function resolveInTime(t) {
+  return new Promise(resolve => {
+    setTimeout(() => {
+      resolve(++yieldorder);
+    }, t*10);
+  });
+}
+
+async function* g() {
+  yield resolveInTime(1);
+  yield ++yieldorder;
+}
+
+var iter = g();
+
+var item1 = iter.next();
+var item2 = iter.next();
+var item3 = iter.next();
+
+async function awaitnexts() {
+  assert.sameValue((await item3).value, undefined)
+  assert.sameValue((await item2).value, 2)
+  assert.sameValue((await item1).value, 1)
```

Comment:
the use of await here is amazing!

Reply:
This is much better than the `next().then(...)` pattern I used in the generated code tests.